File: Amit Kalra/ProjectsViewController.m
//
//  ProjectsViewController.m
//  Amit Nivedan Kalra
//
//  Created by Amit Kalra on 8/13/14.
//  Copyright (c) 2014 Amit Kalra. All rights reserved.
//

/*

#import "ProjectsViewController.h"

@interface ProjectsViewController ()

@end

@implementation ProjectsViewController

//Animated Label

-(IBAction)changeText{
    
    AnimatedLabel.text = @"Amit Nivedan Kalra";
    
}

//Hide Button

-(IBAction)HideButton:(id)sender{
    
    HideButton.hidden = YES;
    AnimatedLabel.hidden = NO;
    
}

//Swipe

-(IBAction)SWIPE:(id)sender {
    
    [self.navigationController popViewControllerAnimated:YES];
    
}


- (void)viewDidLoad {
    
    
    //Scroller
    
    [Scroller setScrollEnabled:YES];
    [Scroller setContentSize:CGSizeMake(320, 5680)];
    
    //Animated Picture
    
    AnimatedPicture.animationImages = [NSArray arrayWithObjects:
                                       [UIImage imageNamed:@"ak.png"],
                                       [UIImage imageNamed:@"San Fran Selfie.jpg"],
                                       [UIImage imageNamed:@"Winter.jpg"],
                                       [UIImage imageNamed:@"Bike pic.jpg"],
                                       [UIImage imageNamed:@"India Selfie.jpg"],
                                       [UIImage imageNamed:@"Walmart fun.jpg"],
                                       [UIImage imageNamed:@"Photo Edit.jpg"],
                                       [UIImage imageNamed:@"Amit old icon.png"], nil];
    
    
    [AnimatedPicture setAnimationRepeatCount:0];
    AnimatedPicture.animationDuration = 3.5;
    [AnimatedPicture startAnimating];

    [super viewDidLoad];
    
	// Do any additional setup after loading the view, typically from a nib.
}

//ROBLOX Game

-(IBAction)ROBLOXGame:(id)sender {
    
    [[UIApplication sharedApplication] openURL: [NSURL URLWithString:@"http://www.roblox.com/Parkour-City-place?id=102726507"]];
    
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


 
 I have a couple of projects that I'm working on..
 
 - Parkour City, a game I started on ROBLOX that I'm bringing to iOS.
 - Creating a crazy fun Santa game for children. (I think it'll be released next year)
 - Kalra Enterprises application for my family buisness, I want to help them out.
 
 
 


@end
 
 */

File: Amit Kalra/ProjectsViewController.swift
//
//  ProjectsViewController.swift
//  Amit Kalra
//
//  Created by Amit Kalra on 4/14/15.
//  Copyright (c) 2015 Amit Kalra. All rights reserved.
//

import UIKit
import AVFoundation

class ProjectsViewController: UIViewController {
    
    //Parkour City
    
    @IBAction func ParkourCity(sender: AnyObject) {
        
        if let PC = NSURL(string:"http://roblox.com/games/102726507/Parkour-City"){
            
            UIApplication.sharedApplication().openURL(PC)
            
        }
        
    }
    
    @IBAction func AppleJumps(sender: AnyObject) {
        
        if let AJ = NSURL(string: "https://itunes.apple.com/us/app/id942130042") {
            
            UIApplication.sharedApplication().openURL(AJ)
        }
    }
    
    
    //Projects Audio
    
    var audioPlayer = AVAudioPlayer()
    
    
    func ProjectsAudio(){
        if let alertSound = NSURL(fileURLWithPath: NSBundle.mainBundle().pathForResource("Projects", ofType: "mp3")!){
            
            AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, error: nil)
            AVAudioSession.sharedInstance().setActive(true, error: nil)
            
            var error:NSError?
            audioPlayer = AVAudioPlayer(data: NSData(contentsOfURL: NSURL(fileURLWithPath: NSBundle.mainBundle().pathForResource("Projects", ofType: "mp3")!)!), error: nil)
            
            audioPlayer.prepareToPlay()
            audioPlayer.play()
            
        }
    }

    
    @IBAction func anotherProjectsAudio(sender: AnyObject) {
        
        ProjectsAudio()
    }
    
    
    //Page Scroller
    
    @IBOutlet weak var scrollView: UIScrollView!
    
    //Animated Background
    
    @IBOutlet var AnimatedBackground: UIImageView!
    
    let backgroundImages = [
        
        UIImage(named: "Blur.jpg")!,
        
        UIImage(named: "Design.jpg")!]
    
    var backgroundIndex = 0
    let backgroundAnimationDuration: NSTimeInterval = 0.25
    let backgroundSwitchingInterval: NSTimeInterval = 2
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        self.scrollView.contentInset = UIEdgeInsetsMake(0, 0, 2900, 0)
        
        //Animated Background
        
        AnimatedBackground.image = backgroundImages[backgroundIndex++]
        animateAnimatedBackground()
        
    }
    
    //Animated Background
    
    func animateAnimatedBackground() {
        CATransaction.begin()
        
        CATransaction.setAnimationDuration(backgroundAnimationDuration)
        CATransaction.setCompletionBlock {
            dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(self.backgroundSwitchingInterval * NSTimeInterval(NSEC_PER_SEC))), dispatch_get_main_queue()) {
                self.animateAnimatedBackground()
            }
        }
        
        let backgroundTransition = CATransition()
        backgroundTransition.type = kCATransitionFade
        /*
        transition.type = kCATransitionPush
        transition.subtype = kCATransitionFromRight
        */
        AnimatedBackground.layer.addAnimation(backgroundTransition, forKey: kCATransition)
        AnimatedBackground.image = backgroundImages[backgroundIndex]
        
        CATransaction.commit()
        
        backgroundIndex = backgroundIndex < backgroundImages.count - 1 ? backgroundIndex + 1 : 0
    }

    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    
}

